### ISSUE TITLE
neural_net_dense_allgather_distributed_mpi tends to predict in a 'safety zone'

### ISSUE THREAD

**Issue by fleapapa** [intel/daal#38]
Hi,

I modified the file, simply to add an 'early stop' function to the code. Nothing else.

Then i trained the 4-layer DNN with 2M data samples. I ran it thru different configurations of nueron numbers. Please visit my [result visualization](https://github.com/fleapapa/WiFi_performance_prediction/blob/master/daal/neural_net_dense_allgather_distributed_mpi2.ipynb).

You could see the predictions tend to stay in a zone between 65 and 75.

I tried to tune initial weight/bias distributions and learning rates (to avoid gradient 'explosion' to NaNs).

You could visit [my github](https://github.com/fleapapa/WiFi_performance_prediction/tree/master/daal) for the code i tested.

Any idea what may be wrong in DAAL code and in mine? 

Thanks,
Peter

**Comment by fleapapa** [comment]
BTW, in contrast, you could visit my [wifi performance predictions](https://github.com/fleapapa/WiFi_performance_prediction) project for visualization of results from using other ML platforms such as sklearn, Spark and Tensorflow. 

I feel visually that the result from DAAL NN is obviously 'flat' when compared with the others.

**Comment by kiselik** [comment]
Hi @fleapapa,

We plan to deprecate Neural Network functionality soon. It means no new features and perfomance turning etc. If you haven't objections, I close the issue
